src/marin/post_training/environments/prime_intellect_env.py

```diff
@@ -0,0 +1,285 @@
+# Copyright 2025 The Marin Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Environment Wrapper for the Environments Hub by Prime-Intellect, which contains a collection of environments.
+https://app.primeintellect.ai/dashboard/environments?ex_sort=most_stars
+"""
+import os
+import json
+import uuid
+import time
+from datetime import datetime
+import logging
+import numpy as np
+import importlib
+import subprocess
+import importlib.util
+from datasets import Dataset
+
+import verifiers as vf
+from typing import cast
+from pathlib import Path
+from verifiers.types import Endpoints
+from verifiers.utils.client_utils import setup_client
+from verifiers.utils.message_utils import messages_to_printable, sanitize_tool_calls
+
+from .marin_env import EnvStep, InferenceContext, MarinEnv
+
+logger = logging.getLogger(__name__)
+
+
+def eval_environment(
+    env: str,
+    env_args: dict,
+    env_dir_path: str,
+    endpoints_path: str,
+    model: str,
+    api_key_var: str,
+    api_base_url: str,
+    num_examples: int,
+    rollouts_per_example: int,
+    max_concurrent: int,
+    max_tokens: int | None,
+    temperature: float | None,
+    sampling_args: dict | None,
+    verbose: bool,
+    save_dataset: bool,
+    save_to_hf_hub: bool,
+    hf_hub_dataset_name: str,
+):
+    logger.setLevel("DEBUG" if verbose else "INFO")
```

Comment:
probably better to remove and just let the user adjust this based on the logger name if desired.